apis/v0/jwt.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v0
+
+import (
+	"fmt"
+
+	"github.com/lestrrat-go/jwx/v2/jwt"
+)
+
+const (
+	// jwtJustificationsKey is the key in the JWT where justifications are stored.
+	// Ideally this would be "justifications", but the RFC and various online
+	// resources recommend key names be as short as possible to keep the JWTs
+	// small. Akamai recommends less than 8 characters and Okta recommends less
+	// than 6.
+	jwtJustificationsKey string = "justs"
+)
+
+// WithTypedJustifications is an option for parsing JWTs that will convert
+// decode the [Justification] claims into the correct Go structure. If this is
+// not supplied, the claims will be "any" and future type assertions will fail.
+func WithTypedJustifications() jwt.ParseOption {
```

Comment:
This is one of the biggest downsides to the lestrrat library. There are globals for registering decoding hooks, but they won't work because of how we have the packages structured.